src/main/java/randoop/main/GenTests.java

```diff
@@ -551,12 +546,15 @@ private static SimpleLog getSimpleLog(String filename) {
   }
 
   /**
-   * Adds patterns for methods to be omitted to the given list. Reads from the {@link
+   * Adds patterns for methods to be omitted to the given list and returns an {@link
+   * OmitMethodsPredicate} using the pattern list. Reads from the {@link
    * GenInputsAbstract#omitmethods_list} file.
    *
    * @param omitmethods the list of {@code Pattern} objects to add new patterns to, must not be null
+   * @return the {@link OmitMethodsPredicate} for the omit patterns from both {@link
+   *     GenInputsAbstract#omitmethods} and {@link GenInputsAbstract#omitmethods_list}.
    */
-  private void extendOmitMethods(List<Pattern> omitmethods) {
+  private OmitMethodsPredicate createOmissionPredicate(List<Pattern> omitmethods) {
```

Comment:
The code in createOmissionPredicate feels out of place, because it is doing two distinct things.  First, it is doing command-line option processing, by setting omitmethods correctly, according to other command-line arguments.  Second, it is creating an OmitMethodsPredicate.  I feel that the first part should appear in GenTests, just after the call to checkOptionsValid(), where other variables are set.  The code would update omitmethods from other variables, so from that point onward omitmethods can be used without referring to the other variables.  This will also accommodate the simplification of OmitMethodsPredicate suggested above.
(Hmm, I wrote that suggestion while reading the new version of the code.  Now that I see the diffs, they indicate that the code used to work that way.  There was a method named extendOmitMethods that you changed into createOmissionPredicate by adding a single line to its end.  I suggest that you reverse the change.)